examples/testgputext.c

```diff
@@ -0,0 +1,393 @@
+#include <SDL3/SDL_main.h>
+#include <SDL3/SDL.h>
+#include <SDL3_ttf/SDL_ttf.h>
+
+#define MATH_3D_IMPLEMENTATION
+#include "math_3d.h"
+
+
+#define MAX_VERTEX_COUNT 4000
+#define MAX_INDEX_COUNT  6000
+
+
+typedef struct Vec2 {
+	float x, y;
+} Vec2;
+
+typedef struct Vec3 {
+	float x, y, z;
+} Vec3;
+
+
+typedef struct Vertex {
+	Vec3 pos;
+	SDL_FColor colour;
+	Vec2 uv;
+} Vertex;
+
+
+typedef struct Context {
+	SDL_GPUDevice *device;
+	SDL_Window *window;
+	SDL_GPUGraphicsPipeline *pipeline;
+	SDL_GPUBuffer *vertex_buffer;
+	SDL_GPUBuffer *index_buffer;
+	SDL_GPUTransferBuffer *transfer_buffer;
+	SDL_GPUSampler *sampler;
+	SDL_GPUCommandBuffer *cmd_buf;
+} Context;
+
+
+typedef struct GeometryData {
+	Vertex *vertices;
+	int vertex_count;
+	int *indices;
+	int index_count;
+} GeometryData;
+
+
+void check_error_bool(const bool res) {
+	if (!res)
```

Comment:
Please use SDL coding style, e.g. 4 spaces, braces on the same line, etc.

Reply:
Is there some script or tool I can use to format it automatically?

Reply:
Yep, you can use clang-format.